bpaf_core/src/split.rs

```diff
@@ -297,7 +305,34 @@ pub(crate) fn split_param<'a>(
     match value {
         OsOrStr::Str(cow) => split_str_param(cow.as_ref(), args, flags),
         OsOrStr::Os(cow) => match ascii_prefix(cow.as_ref()) {
-            Some(_) => todo!(),
+            // SAFETY: cow contains only utf8
+            Some((_, None)) => split_str_param(cow.to_str().unwrap(), args, flags),
+            Some((prefix, Some(suffix))) => match split_str_param(prefix.as_ref(), args, flags)? {
+                Arg::Named { name, value } => {
+                    // we are in the {name}=some-valid-utf8 situation
+                    if let Some(value) = value {
+                        let mut os = value.os();
+                        os.push(suffix.as_ref());
+                        Ok(Arg::Named {
+                            name: name.to_owned(),
+                            value: Some(OsOrStr::from(os)),
+                        })
+                    } else {
+                        Err(Error::fail("Unexpected trailing non-utf8 after a flag"))
+                    }
+                }
+                Arg::ShortSet { .. } => {
+                    Err(Error::fail("Unexpected trailing non-utf8 after a shortset"))
+                }
+                Arg::Positional { value } => {
```

Comment:
Correct, but I think in this case we can take `cow` or `value` we got from the beginning and pass as is. Mostly to avoid re-allocations.

Reply:
Once if we have just the split position and do the slice splitting/resizing outside of the functions.